cmd/juju/firewall/setrule.go

```diff
@@ -52,76 +51,61 @@ func NewSetFirewallRuleCommand() cmd.Command {
 type setFirewallRuleCommand struct {
 	modelcmd.ModelCommandBase
 	modelcmd.IAASOnlyCommand
-	service        string
-	whitelistValue string
+	whitelist string
 
-	whiteList  []string
 	newAPIFunc func() (SetFirewallRuleAPI, error)
 }
 
 // Info implements cmd.Command.
 func (c *setFirewallRuleCommand) Info() *cmd.Info {
-	supportedRules := []string{
-		" -" + string(params.SSHRule),
-	}
 	return jujucmd.Info(&cmd.Info{
 		Name:    "set-firewall-rule",
 		Args:    "<service-name>, --whitelist <cidr>[,<cidr>...]",
 		Purpose: setRuleHelpSummary,
-		Doc:     fmt.Sprintf(setRuleHelpDetails, strings.Join(supportedRules, "\n")),
+		Doc:     fmt.Sprintf(setRuleHelpDetails, deprecationWarning),
 	})
 }
 
 // SetFlags implements cmd.Command.
 func (c *setFirewallRuleCommand) SetFlags(f *gnuflag.FlagSet) {
-	f.StringVar(&c.whitelistValue, "whitelist", "", "list of subnets to whitelist")
+	f.StringVar(&c.whitelist, "whitelist", "", "list of subnets to whitelist")
 }
 
 // Init implements cmd.Command.
 func (c *setFirewallRuleCommand) Init(args []string) (err error) {
 	if len(args) == 1 {
-		c.service = args[0]
-		if c.whitelistValue == "" {
+		if c.whitelist == "" {
 			return errors.New("no whitelist subnets specified")
 		}
-		if err := c.parseCIDRs(&c.whiteList, c.whitelistValue); err != nil {
-			return errors.Annotate(err, "invalid white-list subnet")
+		if args[0] != "ssh" {
+			return errors.NotSupportedf("service %q", args[0])
 		}
-		return nil
 	}
 	if len(args) == 0 {
 		return errors.New("no well known service specified")
 	}
 	return cmd.CheckEmpty(args[1:])
 }
 
-func (c *setFirewallRuleCommand) parseCIDRs(cidrs *[]string, value string) error {
```

Comment:
This is actually useful for early erroring out, I think we should keep this, or similar to this.